### PR TITLE
Persist metadata in Data Explorer

### PR DESCRIPTION
**Pull request by emeeks**
* Honor dx metadata
* Fire an onMetadataChange fn when metadata changes
* Rudimentary tests to ensure the component works without metadata

### CODE REVIEW

**Comment by emeeks** [comment]
Whoops looks like I forgot to update the test snapshots.

File: packages/transform-dataresource/src/index.js [hunk 1]
```diff
@@ -190,23 +220,21 @@ class DataResourceTransform extends React.Component<Props, State> {
...
-  //CHART CHOICES
-  //DERIVED DATA
-
-  shouldComponentUpdate(): boolean {
-    return true;
+  componentWillMount() {
+    // This is necessary to render any charts based on passed metadata because the grid doesn't result from the updateChart function but any other view does
+    if (this.state.view !== "grid") {
+      this.updateChart(this.state);
+    }
```
Comment (BenRussert) [inline packages/transform-dataresource/src/index.js:239]: Is there a reason we need grid to always be the default initial view? I wondered in the past if the view should just be a prop with "grid" being default. Updating after mounting will essentially do the same thing, but I'm wondering if I understand why grid has to be the initial view in the constructor.

File: packages/transform-dataresource/src/index.js [hunk 2]
```diff
@@ -20,17 +20,18 @@ type dataProps = {
...
 
+type AreaType = "hexbin" | "heatmap" | "contour";
+
+type SummaryType = "violin" | "joy" | "histogram" | "heatmap" | "boxplot";
+
+type PieceType = "bar" | "point" | "swarm" | "clusterbar";
+
+type HierarchyType = "dendrogram" | "treemap" | "partition";
+
+type NetworkType = "force" | "sankey" | "arc" | "matrix";
```
Comment (BenRussert) [inline packages/transform-dataresource/src/index.js:33]: I really like having these types declared up here personally.

File: packages/transform-dataresource/src/index.js [hunk 3]
```diff
@@ -42,25 +43,51 @@ export type View =
...
+};
+
+type Props = {
+  data: dataProps,
+  metadata: { dx: dxMetaProps },
+  theme?: string,
+  expanded?: boolean,
+  height?: number,
+  mediaType: "application/vnd.dataresource+json",
+  onMetadataChange?: Function
```
Comment (BenRussert) [inline packages/transform-dataresource/src/index.js]: `onMetaDataChange?: ({ dx: dxMetaProps }) => void`, right?

**Review by BenRussert — APPROVED** [review]
This is really nice, and we even have tests now! I put some other comments/questions below.

File: packages/transform-dataresource/src/index.js (hunk 1 above)
Comment (emeeks) [reply packages/transform-dataresource/src/index.js:239]: Oh I like the whole prop with grid as default, I'll add that.

File: packages/transform-dataresource/src/index.js (hunk 2 above)
Comment (emeeks) [reply packages/transform-dataresource/src/index.js:33]: Yeah, I'm still getting used to the whole "More than me needs to understand this stuff". The "arc" and "matrix" support is coming soon BTW but it's not in there yet.

File: packages/transform-dataresource/src/index.js (hunk 3 above)
Comment (emeeks) [reply packages/transform-dataresource/src/index.js]: Oh good point. Excellent!

**Pull request merged**